### PR TITLE
Feature/av2 926/update reset password api call

### PR DESCRIPTION
**Pull request by sasangachathumal**
**Story/Ticket Reference ID from JIRA:**
https://intersective.atlassian.net/browse/AV2-926

**Additional Comments:**
`Add here`

**Checklist:**

- [x] Unit test completed?: (Y)

All checklists are okay.

This PR looks great - it's ready to merge! Please review and let's ship it. :)

### CODE REVIEW

File: src/app/auth/auth.service.ts [hunk 1]
```diff
@@ -20,13 +20,13 @@ const api = {
   verifyRegistration: 'api/verification_codes.json',
   register: 'api/registration_details.json',
   verifyResetPassword: 'api/auths.json?action=verify_reset_password',
```
Comment (shawnm0705) [inline src/app/auth/auth.service.ts]: If we don't need this anymore, please delete it

**Review by trtshen — APPROVED** [review]

**Review by shawnm0705 — CHANGES_REQUESTED** [review]

File: src/app/shared/request/request.service.ts [hunk 2]
```diff
@@ -98,16 +98,16 @@ export class RequestService {
...
     if (isLoginAPI) {
-      endpointUrl = this.loginApiUrl + endpoint;
+      endpointUrl = this.utils.urlFormatter(this.loginApiUrl, endpoint);
     } else if (this.storage.stackConfig && this.storage.stackConfig.coreApi) {
-      endpointUrl = this.storage.stackConfig.coreApi + endpoint;
+      endpointUrl = this.utils.urlFormatter(this.storage.stackConfig.coreApi, endpoint);
     } else {
       throw new Error('Can not find API URL.');
     }
     if (endpoint.includes('https://') || endpoint.includes('http://')) {
```
Comment (shawnm0705) [inline src/app/shared/request/request.service.ts]: since we have the urlFormatter, we don't need to check the http here anymore

**Review by shawnm0705 — CHANGES_REQUESTED** [review]

File: src/app/shared/request/request.service.ts (hunk 2 above)
Comment (sasangachathumal) [reply src/app/shared/request/request.service.ts]: @shawnm0705 
in some places we pass full url as endpoint ( "http://abcd.com/123" ). like the URL we use to get user location ("https://ipapi.co/json"). this is for check that. to not add our API URL in to those urls.
if endpoint have http or https then not adding any url to endpoint.

Comment (shawnm0705) [reply src/app/shared/request/request.service.ts]: inside the urlFormatter, we also checks if it has `http`, so this is redundant I think?

**Review by shawnm0705 — CHANGES_REQUESTED** [review]

File: src/app/shared/request/request.service.ts (hunk 2 above)
Comment (sasangachathumal) [reply src/app/shared/request/request.service.ts]: I can't find any place we use this like this. let's remove this as you mention, if there any issue came we can fix this later

**Review by shawnm0705 — APPROVED** [review]

**Pull request merged**